modules/QnA/resources/qna.js

```diff
@@ -0,0 +1,200 @@
+(function (mw, $) {
+	"use strict";
+
+	mw.PluginManager.add('qna', mw.KBasePlugin.extend({
+
+		defaultConfig: {
+			templatePath: '../QnA/resources/qna.tmpl.html',
+			cssFileName: 'modules/QnA/resources/qna.css'
+		},
+
+		getBaseConfig: function() {
+			var parentConfig = this._super();
+			return $.extend({}, parentConfig, {
+				qnaTargetId: null
+			});
+		},
+
+		setup: function () {
+			this.addBindings();
+		},
+
+		addBindings: function () {
+			var _this = this;
+			var embedPlayer = this.getPlayer();
+
+			this.bind('layoutBuildDone', function (event, screenName) {
+				// add the Q&A toggle button to be on the video
+				embedPlayer.getVideoHolder().append('<div class="qna-on-video-btn icon-qna-close"></div>');
+				_this.getQnaContainer();
+				var qnaObject =  $(window['parent'].document.getElementById(embedPlayer.id )).parent().find( ".qnaInterface" );
```

Comment:
we must check that the player is not in unfriendly iframe - this code will break if we do